backend/asset/s3_videos.go

```diff
@@ -0,0 +1,70 @@
+package asset
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ansel1/merry/v2"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+)
+
+func getSignedAudiolessVideoURLForAssetID(context context.Context, services externalServices, config config, assetID int32) (*string, error) {
+	queries := services.GetQueries()
+
+	storagePath, err := queries.GetAssetStoragePath(context, assetID)
+	if err != nil {
+		return nil, merry.Wrap(err)
+	}
+
+	if storagePath.String == "" {
+		return nil, merry.New("no storage path found for assetID", merry.WithUserMessage("no storage path found for assetID: "+string(assetID)))
+	}
+
+	s3client := services.GetS3Client()
+	objects, err := s3client.ListObjectsV2(context, &s3.ListObjectsV2Input{
+		Bucket: config.GetStorageBucket(),
+		Prefix: &storagePath.String,
+	})
+	if err != nil {
+		return nil, merry.Wrap(err)
+	}
+
+	if len(objects.Contents) == 0 {
+		return nil, merry.New("no objects found", merry.WithUserMessage("no objects found for assetID: "+string(assetID)))
+	}
```

Comment:
I think you can drop this section, because the loop doesn't care if the len == 0 and you check afterwards anyways.
```suggestion
```